utils/src/main/kotlin/Utils.kt

```diff
@@ -380,6 +380,38 @@ fun File.toSafeURI(): URI {
     return URI("file", "", fileUri.path, fileUri.query, fileUri.fragment)
 }
 
+/**
+ * Populates and returns the [destination] mutable map with key-value pairs provided by [transform] function applied to
+ * each element of the given collection and the sequential index.
+ *
+ * If any of two pairs would have the same key the last one gets added to the map.
+ */
+inline fun <T, K, V, M : MutableMap<in K, in V>> Iterable<T>.associateToIndexed(
```

Comment:
I wonder whether `associateIndexedTo` is a better name. I think it's more in line with the naming convention of these functions to have the `To` at the end, since it refers to the first argument. But I can't find an example right now.

Reply:
You are right, there is e.g. `filterTo` and `filterIndexedTo`, so I adapted to this:
https://github.com/JetBrains/kotlin/blob/bea94d1d5945c57e2999531b0ed5df6653133805/libraries/stdlib/common/src/generated/_Collections.kt